### ISSUE TITLE
Example01 fails to compile on GCC 4.8.5

### ISSUE THREAD

**Issue by ryanlevy** [aeantipov/gftools#2]
`example01.cpp` fails to compile on my machine. The compiler error in question:
```
In file included from /Users/user/UMICH/software/gftools/./gftools/grid_object.hpp:281:0,
                 from /Users/user/UMICH/software/gftools/example/example01.cpp:12:
/Users/user/UMICH/software/gftools/./gftools/grid_object.hxx: In instantiation of 'gftools::grid_object_base<ContainerType, GridTypes>::grid_object_base(const gftools::grid_object_base<ContainerType, GridTypes>&) [with ContainerType = gftools::container<double, 1ul>; GridTypes = {gftools::real_grid}]':
/Users/user/UMICH/software/gftools/./gftools/grid_object.hpp:240:108:   required from 'gftools::grid_object<typename ContainerType::value_type, GridTypes ...> gftools::grid_object_base<ContainerType, GridTypes>::operator+(const gftools::grid_object_base<ContainerType, GridTypes>&) const [with ContainerType = gftools::container<double, 1ul>; GridTypes = {gftools::real_grid}; gftools::grid_object<typename ContainerType::value_type, GridTypes ...> = gftools::grid_object_base<gftools::container<double, 1ul>, gftools::real_grid>; typename ContainerType::value_type = double]'
/Users/user/UMICH/software/gftools/example/example01.cpp:72:30:   required from here
/Users/user/UMICH/software/gftools/./gftools/grid_object.hxx:61:20: error: use of deleted function 'gftools::container<double, 1ul>::container(const gftools::container<double, 1ul>&)'
     tail_(rhs.tail_)
                    ^
In file included from /Users/user/UMICH/software/gftools/example/example01.cpp:10:0:
/Users/user/UMICH/software/gftools/./gftools/container.hpp:244:8: note: 'gftools::container<double, 1ul>::container(const gftools::container<double, 1ul>&)' is implicitly declared as deleted because 'gftools::container<double, 1ul>' declares a move constructor or move assignment operator
 struct container : container_base<ValueType,N,typename boost::multi_array<ValueType, N>> {
        ^

```

My GCC version

```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/opt/local/libexec/gcc/x86_64-apple-darwin14/4.8.5/lto-wrapper
Target: x86_64-apple-darwin14
Configured with: /opt/local/var/macports/build/_opt_mports_dports_lang_gcc48/gcc48/work/gcc-4.8.5/configure --prefix=/opt/local --build=x86_64-apple-darwin14 --enable-languages=c,c++,objc,obj-c++,lto,fortran,java --libdir=/opt/local/lib/gcc48 --includedir=/opt/local/include/gcc48 --infodir=/opt/local/share/info --mandir=/opt/local/share/man --datarootdir=/opt/local/share/gcc-4.8 --with-local-prefix=/opt/local --with-system-zlib --disable-nls --program-suffix=-mp-4.8 --with-gxx-include-dir=/opt/local/include/gcc48/c++/ --with-gmp=/opt/local --with-mpfr=/opt/local --with-mpc=/opt/local --with-isl=/opt/local --disable-isl-version-check --with-cloog=/opt/local --disable-cloog-version-check --enable-stage1-checking --disable-multilib --enable-lto --enable-libstdcxx-time --with-as=/opt/local/bin/as --with-ld=/opt/local/bin/ld --with-ar=/opt/local/bin/ar --with-bugurl=https://trac.macports.org/newticket --with-pkgversion='MacPorts gcc48 4.8.5_0'
Thread model: posix
gcc version 4.8.5 (MacPorts gcc48 4.8.5_0) 
```

**Comment by aeantipov** [comment]
Ryan, this should be fixed with 0f9a992 
Check if that works.

**Comment by ryanlevy** [comment]
This now compiles (but fails due to issue #3).
